src/actions/attack.cpp

```diff
@@ -932,18 +932,18 @@ void attack::fire_event(const std::string& n)
 
 	if(a_stats_->weapon != nullptr && a_.valid()) {
 		if(d_stats_->weapon != nullptr && d_.valid()) {
-			a_ctx.emplace(a_stats_->weapon->specials_context(a_.get_unit(), d_.get_unit(), a_.loc_, d_.loc_, true, d_stats_->weapon));
+			a_ctx.emplace(a_stats_->weapon->specials_context(nullptr, nullptr, a_.loc_, d_.loc_, true, d_stats_->weapon));
```

Comment:
...which _might_ mean you need a new `specials_context()` overload to cover this case, but uh... I think it actually makes no sense to be passing `nullptr` here? Before, it was passing a valid unit. Now, it's not passing a unit. That seems wrong.